entity-framework/core/querying/related-data.md

```diff
@@ -109,6 +109,27 @@ Contents of `School` navigation of all People who are Students can be eagerly lo
   context.People.Include("School").ToList()
   ```
 
+### Filtered include
+
+> [!NOTE]
+> This feature was introduced in EF Core 5.0.
+
+Filtered include feature enables applying some operations directly on the included collection navigation, which allows for filtering and sorting of the results.
+
+Supported operations are: `Where`, `OrderBy`, `OrderByDescending`, `ThenBy`, `ThenByDescending`, `Skip`, and `Take`.
+
+Those operations should be applied inside the Include method:
+
+[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs#FilteredInclude)]
+
+Each included navigation allows only one unique set of filter operations. In case when Include operation is applied more than once on a given collection navigation (`blog.Posts` in the examples below), filter operations should only be applied on one of them. 
+
+[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs#MultipleLeafIncludesFiltered1)]
+
+Alternatively, identical operations can be applied for each navigation that is included multiple times.
+
+[!code-csharp[Main](../../../samples/core/Querying/RelatedData/Sample.cs#MultipleLeafIncludesFiltered2)]
+
```

Comment:
many of the filtered include scenarios translate to apply, so sqlite will throw - should we add a warning/caution for that here?

Reply:
for filtered includes, take works in sqlite right?

Reply:
Nope - 

```
SELECT "l"."Id", "l"."Date", "l"."Name", "l"."OneToMany_Optional_Self_Inverse1Id", "l"."OneToMany_Required_Self_Inverse1Id", "l"."OneToOne_Optional_Self1Id", "t"."Id", "t"."Date", "t"."Level1_Optional_Id", "t"."Level1_Required_Id", "t"."Name", "t"."OneToMany_Optional_Inverse2Id", "t"."OneToMany_Optional_Self_Inverse2Id", "t"."OneToMany_Required_Inverse2Id", "t"."OneToMany_Required_Self_Inverse2Id", "t"."OneToOne_Optional_PK_Inverse2Id", "t"."OneToOne_Optional_Self2Id"
FROM "LevelOne" AS "l"
OUTER APPLY (
    SELECT "l0"."Id", "l0"."Date", "l0"."Level1_Optional_Id", "l0"."Level1_Required_Id", "l0"."Name", "l0"."OneToMany_Optional_Inverse2Id", "l0"."OneToMany_Optional_Self_Inverse2Id", "l0"."OneToMany_Required_Inverse2Id", "l0"."OneToMany_Required_Self_Inverse2Id", "l0"."OneToOne_Optional_PK_Inverse2Id", "l0"."OneToOne_Optional_Self2Id"
    FROM "LevelTwo" AS "l0"
    WHERE "l"."Id" = "l0"."OneToMany_Optional_Inverse2Id"
    ORDER BY "l0"."Name"
    LIMIT 3
) AS "t"
ORDER BY "l"."Id", "t"."Name", "t"."Id"
```

Where works, OrderBy.AsQueryable works, Skip/Take fails due to APPLY, naked OrderBy fails due to a bug

Reply:
Something is wrong then. We added whole support for RowNumberExpression to convert apply to join when there is only take. https://github.com/dotnet/efcore/pull/17293
There is also test which converts Take(4) in select many. At the minimum query you posted above has nothing in it which stops it from getting converted to join.

Also related to it: If we fix https://github.com/dotnet/efcore/issues/17326, then skip would also convert to number and most of it should get translated to Sqlite. (For filtered include case, everything would work).

Reply:
will investigate what's up

Reply:
```
ctx.LevelOne.SelectMany(l1 => l1.OneToMany_Optional1.OrderBy(x => x.Name).Take(3).ToList()).ToList();
```

still translates to apply on sqlite (and fails).

Reply:
File an issue and don't add any warning here.